### PR TITLE
242 implement pivotssh exec function

### LINKED ISSUE #242: Implement `pivot.ssh_exec` function
**Is your feature request related to a problem? Please describe.**
This feature will allow eldritch to connect to remote hosts over ssh.

**Describe the solution you'd like**
- Implement a function that will take a user name, password / priv key, host, and port.
- Connect to the remote SSH server and run a command string similar to `sys.shell`

**Describe alternatives you've considered**
N/a

**Additional context**
N/a

### PR DESCRIPTION
**Pull request by hulto**
#### What this PR does / why we need it:
This PR implements an SSH exec function allowing eldritch to run commands on a remote host over ssh.

#### Which issue(s) this PR fixes:
Fixes #242

### CODE REVIEW

File: implants/lib/eldritch/src/pivot.rs [hunk 1]
```diff
@@ -96,4 +103,105 @@ fn methods(builder: &mut MethodsBuilder) {
...
+impl Session {
+    async fn connect(
+        user: String,
+        password: Option<String>,
+        key: Option<String>,
+        key_password: Option<&str>,
+        addrs: String,
+    ) -> anyhow::Result<Self> {
+        let config = client::Config {
+            ..<_>::default()
```
Comment (Cictrone) [inline implants/lib/eldritch/src/pivot.rs:138]: wut on God's green earth is going on here?

File: implants/lib/eldritch/src/pivot/ssh_exec_impl.rs [hunk 2]
```diff
@@ -1,5 +1,205 @@
...
+    stdout: String,
+    status: i32,
+}
+
+async fn handle_ssh_exec(target: String, port: u16, command: String, username: String, password: Option<String>, key: Option<String>, key_password: Option<&str>, timeout: Option<u32>) -> Result<SSHExecOutput> {
+    let mut ssh = tokio::time::timeout(
+        std::time::Duration::from_secs(timeout.unwrap_or(3).into()),
+        Session::connect(username, password, key, key_password, format!("{}:{}", target, port)),
+
+    ).await??;
```
Comment (Cictrone) [inline implants/lib/eldritch/src/pivot/ssh_exec_impl.rs:16]: the rare double `?`

File: implants/lib/eldritch/src/pivot/ssh_exec_impl.rs [hunk 3]
```diff
@@ -1,5 +1,205 @@
...
+    use std::sync::{Mutex, Arc};
+    use russh::*;
+    use russh::server::{Auth, Session, Msg};
+    use russh_keys::*;
+    use std::collections::HashMap;
+    use super::*;
+
+    // SSH Server utils
+    #[derive(Clone)]
+    #[allow(dead_code)]
```
Comment (Cictrone) [inline implants/lib/eldritch/src/pivot/ssh_exec_impl.rs:71]: the sussiest baka

Comment (Cictrone) [reply implants/lib/eldritch/src/pivot/ssh_exec_impl.rs:71]: why dis here?

File: implants/lib/eldritch/src/pivot/ssh_exec_impl.rs [hunk 4]
```diff
@@ -1,5 +1,205 @@
...
+
+        async fn channel_open_session(self, channel: Channel<Msg>, session: Session) -> Result<(Self, bool, Session), Self::Error> {
+            {
+                let mut clients = self.clients.lock().unwrap();
+                clients.insert((self.id, channel.id()), channel);
+            }
+            Ok((self, true, session))
+        }
+
+        #[allow(unused_variables)]
```
Comment (Cictrone) [inline implants/lib/eldritch/src/pivot/ssh_exec_impl.rs:99]: sus

File: implants/lib/eldritch/src/pivot/ssh_exec_impl.rs [hunk 5]
```diff
@@ -1,5 +1,205 @@
...
+            let tmp_res = Command::new(command_string)
+                .args(command_args)
+                .output()
+                .expect("failed to execute process");
+            session.data(channel, CryptoVec::from(tmp_res.stdout));
+            session.close(channel); // Only gonna send one command.
+            Ok((self, session))
+        }
+
+        #[allow(unused_variables)]
```
Comment (Cictrone) [inline implants/lib/eldritch/src/pivot/ssh_exec_impl.rs:133]: sus

File: implants/lib/eldritch/src/pivot/ssh_exec_impl.rs [hunk 6]
```diff
@@ -1,5 +1,205 @@
...
+        async fn auth_publickey(self, _: &str, _: &key::PublicKey) -> Result<(Self, Auth), Self::Error> {
+            Ok((self, server::Auth::Accept))
+        }
+
+        #[allow(unused_variables)]
```
Comment (Cictrone) [inline implants/lib/eldritch/src/pivot/ssh_exec_impl.rs:138]: sus

**Review by Cictrone — APPROVED** [review]
looks fine, just alot of questions and sus.

File: implants/lib/eldritch/src/pivot/ssh_exec_impl.rs (hunk 3 above)
Comment (hulto) [reply implants/lib/eldritch/src/pivot/ssh_exec_impl.rs:71]: The ssh library boiler plate has a bunch of sections that while actually used lint as dead_code. Here `client_pubkey` gets linted as unused.

File: implants/lib/eldritch/src/pivot/ssh_exec_impl.rs (hunk 4 above)
Comment (hulto) [reply implants/lib/eldritch/src/pivot/ssh_exec_impl.rs:99]: Yeah same deal ^ library's boiler plate is kinda scuffed.
I don't think any code is actually unused but that's how it lints out.

File: implants/lib/eldritch/src/pivot/ssh_exec_impl.rs (hunk 5 above)
Comment (hulto) [reply implants/lib/eldritch/src/pivot/ssh_exec_impl.rs:133]: ^^

File: implants/lib/eldritch/src/pivot/ssh_exec_impl.rs (hunk 6 above)
Comment (hulto) [reply implants/lib/eldritch/src/pivot/ssh_exec_impl.rs:138]: ^^

File: implants/lib/eldritch/src/pivot.rs (hunk 1 above)
Comment (hulto) [reply implants/lib/eldritch/src/pivot.rs:138]: This is I believe similar to the javascript "the rest" notation `...` and is how you can tell a rust type to populate the rest of a structs fields with the defaults.

File: implants/lib/eldritch/src/pivot/ssh_exec_impl.rs (hunk 2 above)
Comment (hulto) [reply implants/lib/eldritch/src/pivot/ssh_exec_impl.rs:16]: Errors are caught in the parent function `ssh_exec`

**Pull request merged**